Framework/API/src/FunctionProperty.cpp

```diff
@@ -71,13 +75,25 @@ Json::Value FunctionProperty::valueAsJson() const { return Json::Value(value());
  */
 std::string FunctionProperty::getDefault() const { return ""; }
 
+/** Returns true if this property is optional.
+ *  @return true if this property is optional.
+ */
+bool FunctionProperty::isOptional() const { return (m_optional == PropertyMode::Optional); }
+
 /** Set the function definition.
  *  Also tries to create the function with FunctionFactory.
  *  @param value :: The function definition string.
  *  @return Error message from FunctionFactory or "" on success.
  */
 std::string FunctionProperty::setValue(const std::string &value) {
   std::string error;
+
+  if (isOptional() && value.empty()) {
+    m_value = std::shared_ptr<IFunction>();
+    m_definition = value;
+    return error;
```

Comment:
is `error` just an empty string?

Reply:
Yes in this case there is no error because the function is optional and is left empty

Reply:
ok

Reply:
Could we put a comment with words to that effect so someone doesn't add "the missing string" back